alica_engine/src/engine/collections/Variant.cpp

```diff
@@ -3,7 +3,43 @@
 namespace alica
 {
 
-constexpr size_t Variant::kUnionSize;
-constexpr size_t Variant::kVariantSize;
+void variant::serializeTo(std::string& arr, const Variant& var)
+{
+    arr.clear();
+    arr += std::to_string(var.index()); // type
+
+    std::visit( [&](auto&& arg) {
+        using T = std::decay_t<decltype(arg)>;
+        if constexpr (!(std::is_same_v<std::monostate, T>)) {
+            arr += (std::to_string(arg));
+        }
+    }, var );
+}
+
+void variant::loadFrom(const std::string& arr, Variant& var)
+{
+    if (arr.size() < 2) { // should atleast contain once char for type and one char for the data
+        var.emplace<std::monostate>();
+        return;
+    }
+
+    auto type = std::stoi(arr.substr(0, 1)); // get type of the data
+    std::string data = arr.substr(1); // get the data
```

Comment:
Can we somehow use string_view here?  Copying seems unnecessary

Reply:
Looks like the conversions aren't pretty with string view and if we want to avoid copy its best to just call the conversion functions that take a c style str.  Not worth the effort atm I think so we can optimize this later